### PR TITLE
fix: Resend invite operation on users list

### PR DESCRIPTION
**Pull request by RicardoE105**
## Summary

Pending users were added into the store with the property `globalRole`, but when we return the users we save them in the store with the property `role` instead.

image

## Before

See video on ticket -> https://linear.app/n8n/issue/HELP-645/resend-invitation-link-does-not-work-anymore

## Now

CleanShot 2024-10-22 at 16 10 25

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/HELP-645/resend-invitation-link-does-not-work-anymore

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

**Review by netroy — DISMISSED** [review]
since `users.store` already has a test file, can we please add a test case to cover for this? that way we can avoid the same regression in the future.

File: packages/editor-ui/src/stores/users.store.test.ts [hunk 1]
```diff
@@ -58,4 +63,29 @@ describe('users.store', () => {
...
 		});
 	});
+
+	describe('inviteUsers', () => {
+		it('should add pending user to the store', async () => {
+			const usersStore = useUsersStore();
+
+			inviteUsers.mockResolvedValueOnce([
+				{
+					user: { role: 'global:member', email: 'test@n8n.io', emailSent: true },
```
Comment (despairblue) [inline packages/editor-ui/src/stores/users.store.test.ts:73]:
```suggestion
					user: { id: 'random-id', email: 'test@n8n.io', emailSent: true },
```

The API does not return the role, but the id.

File: packages/editor-ui/src/stores/users.store.test.ts [hunk 2]
```diff
@@ -58,4 +63,29 @@ describe('users.store', () => {
...
+				},
+			]);
+
+			await usersStore.inviteUsers([{ email: 'test@n8n.io', role: 'global:admin' }]);
+
+			expect(usersStore.allUsers[0]).toMatchObject(
+				expect.objectContaining({
+					role: 'global:member',
```
Comment (despairblue) [inline packages/editor-ui/src/stores/users.store.test.ts:81]:
```suggestion
					role: 'global:admin',
```

**Review by despairblue — CHANGES_REQUESTED** [review]
The test wasn't preventing a regression. The API mock contained the role, but in reality this happens because the API does not return the role, right? 

Which begs the question if the store code could be simplified by changing the API?

File: packages/editor-ui/src/stores/users.store.test.ts (hunk 2 above)
Comment (RicardoE105) [reply packages/editor-ui/src/stores/users.store.test.ts:81]: Why do we need this?/

**Comment by RicardoE105** [comment]
@despairblue yes, added the role there is an overlooked. Just committed your change. With the ID and not returning the role still works.

File: packages/editor-ui/src/stores/users.store.test.ts (hunk 2 above)
Comment (despairblue) [reply packages/editor-ui/src/stores/users.store.test.ts:81]: On line 77 it says the role is `global:admin`. We can also change line 77 to `global:member`.

**Comment by despairblue** [comment]
> @despairblue yes, adding the role there is an overlooked. Just committed your change. Adding the the ID and not returning the role still works.

Would it be cleaner to add the `role` property to the API response?

**Comment by despairblue** [comment]
@RicardoE105 this is what I mean:

https://github.com/n8n-io/n8n/commit/a6565d3cbf869ae9dba025238e6650dbbfb485e0

Just wondering if that is cleaner, than having the FE retain data it sends to the API and merge it with the response.

**Comment by RicardoE105** [comment]
@despairblue gotcha now. Yet let's do that. Will include that change if this PR if you do not mind.

**Review by despairblue — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.65.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.65.0)

**Pull request merged**